pkg/config/remote/meta/meta.go

```diff
@@ -8,70 +8,90 @@ package meta
 
 import (
 	_ "embed"
+	"encoding/json"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/go-tuf/data"
 )
 
 var (
-	//go:embed prod.1.director.json
-	prodRootDirector1 []byte
-	//go:embed prod.1.config.json
-	prodRootConfig1 []byte
+	//go:embed prod.director.json
+	prodRootDirector []byte
+	//go:embed prod.config.json
+	prodRootConfig []byte
 
-	//go:embed staging.1.director.json
-	stagingRootDirector1 []byte
-	//go:embed staging.1.config.json
-	stagingRootConfig1 []byte
+	//go:embed staging.director.json
+	stagingRootDirector []byte
+	//go:embed staging.config.json
+	stagingRootConfig []byte
 )
 
-// EmbeddedRoot is an embedded root
-type EmbeddedRoot []byte
-
-// EmbeddedRoots is a map of version => EmbeddedRoot
-type EmbeddedRoots map[uint64]EmbeddedRoot
-
-var (
-	prodRootsDirector = EmbeddedRoots{1: prodRootDirector1}
-	prodRootsConfig   = EmbeddedRoots{1: prodRootConfig1}
+// EmbeddedRoot is an embedded root with its version parsed
+type EmbeddedRoot struct {
+	latest uint64
+	root   []byte
+}
 
-	stagingRootsDirector = EmbeddedRoots{1: stagingRootDirector1}
-	stagingRootsConfig   = EmbeddedRoots{1: stagingRootConfig1}
-)
```

Comment:
Why not just update these to `EmbeddedRoots{N: stagingRootConfigN}`? The structure makes it look like it was made for this exact purpose, and it avoids an unnecessary deserialisation at every agent startup.

Reply:
Where would be the second place? It looks like an update would just be:
1. Change file
2. Bump number in the EmbeddedRoots struct

Which isn't a lotta work tbh; and if you're worried about desync a unit test is more than enough